products/compute/api.yaml

```diff
@@ -11536,83 +11542,185 @@ objects:
         name: 'defaultService'
         resource: 'BackendService'
         imports: 'selfLink'
-        description:
-          A reference to BackendService resource if none of the hostRules match.
-        required: true
+        description: |
+          The BackendService resource to which traffic is
+          directed if none of the hostRules match. If defaultRouteAction is additionally
+          specified, advanced routing actions like URL Rewrites, etc. take effect prior to
+          sending the request to the backend. However, if defaultService is specified,
+          defaultRouteAction cannot contain any weightedBackendServices. Conversely, if
+          routeAction specifies any weightedBackendServices, service must not be
+          specified. Only one of defaultService, defaultUrlRedirect  or
+          defaultRouteAction.weightedBackendService must be set.
       - !ruby/object:Api::Type::String
         name: 'description'
         description: |
-          An optional description of this resource. Provide this property when
-          you create the resource.
-      # 'fingerprint' used internally for object consistency.
+          An optional description of this resource. Provide this property when you create
+          the resource.
+      - !ruby/object:Api::Type::Integer
+        name: 'id'
+        description: 'The unique identifier for the resource.'
+        output: true
+      - !ruby/object:Api::Type::Fingerprint
+        name: 'fingerprint'
+        description: |
+          Fingerprint of this resource. A hash of the contents stored in this object. This
+          field is used in optimistic locking.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'headerAction'
```

Comment:
Can you apply the changes you made to `headerAction` to the other occurrences as well?